deployment/aws-python-storage-stack-automation/storage_stack_lifecycle.yaml

```diff
@@ -352,6 +352,243 @@ Resources:
+        - implodelambdarole
+        - Arn
+      Code:
+        ZipFile: |
+          import json
+          import os
+          import boto3
+          import urllib3
+          from botocore.exceptions import ClientError
+          http = urllib3.PoolManager()
+
+          fss_key = os.environ["WS_API"]
+          ext_id_url = "https://cloudone.trendmicro.com/api/filestorage/external-id"
+          stacks_url = "https://cloudone.trendmicro.com/api/filestorage/stacks"
+          delete_url = "https://cloudone.trendmicro.com/api/filestorage/stacks/"
+
+          def lambda_handler(event, context):
+              # get secret
+              secrets = boto3.client('secretsmanager').get_secret_value(SecretId=fss_key)
+              sm_data = json.loads(secrets["SecretString"])
+              ws_key = sm_data["wsapikey"]
+
+              #gather bucket name from event
+              bucket_name = event['detail']['requestParameters']['bucketName']
+              print("S3 Bucket: " + bucket_name)
+              
+              # gather cloud one ext id
+              r = http.request(
+                  "GET",
+                  ext_id_url,
+                  headers={
+                      "Content-Type": "application/json",
+                      "api-secret-key": ws_key,
+                      "Api-Version": "v1",
+                    },
+                )
+              ext = json.loads(r.data.decode("utf-8"))
+              ext_id = ext["externalID"]
+              print("CloudOne ExtID: " + ext_id)
+              
+              # gather aws account ID
+              account_id = event["account"]
+              print("AWS account ID: " + account_id)
+
+              #gather stack id
+              has_next_value = False
+              next = ""
+              #default limit is 25
+              q_params = {
+                "provider": "aws",
+                "status": "ok",
+              }
+              id_call = http.request('GET', stacks_url, q_params, headers = {'Content-Type': 'application/json', 'api-secret-key': ws_key, 'Api-Version': 'v1'})
+              try:
+                  id_resp = json.loads(id_call.data.decode('utf-8'))
+              except json.decoder.JSONDecodeError:
+                  time.sleep(1)
+                  id_resp = json.loads(id_call.data.decode('utf-8'))
+              for data in id_resp['stacks']:
+                  if 'storage' in data and data['storage'] is not None:
+                      if bucket_name == data['storage']:
+                          stack_name = data['name']
+                          stack_id = data['stackID']
+                          print(stack_name)
+                          print(stack_id)
+                          stack_removal(ws_key, stack_id,stack_name)
+                      else:
+                        if "next" in id_resp:
+                          has_next_value = True
+                          next = id_resp['next']
+                        while has_next_value:
```

Comment:
It's incorrect to handle the `next` in the else block because then for every stack in id_resp, it will check `next`. It should instead be after the loop at line#515.

Reply:
Thank you for letting me know.